packages/components/src/components/input/input.stories.ts

```diff
@@ -0,0 +1,676 @@
+              name: 'right',
+              value: '<sd-icon slot="right" library="global-resources" name="system/picture"></sd-icon>'
+            }
+          ],
+          args
+        })
+      )}
+    `;
+  }
+};
+
+/**
+ * Use the `form-control`, `form-control-label`, `form-control-input`, `form-control-help-text`, `base`, `border`, `input`, `left`, `clear-button`, and `right` part selectors to customize the input.
+ */
+
+export const Parts = {
+  parameters: {
+    controls: {
+      exclude: ['label', 'left', 'right', 'clear-icon', 'help-text', 'clearable', 'value']
+    }
+  },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'template',
+          name: 'sd-input::part(...){outline: solid 2px red}',
+          values: [
+            'form-control',
+            'form-control-label',
+            'form-control-input',
+            'form-control-help-text',
+            'base',
+            'border',
+            'input',
+            'left',
+            'clear-button',
+            'right'
+          ].map(part => {
+            return {
+              title: part,
+              value: `<style>#part-${part} sd-input::part(${part}){outline: solid 2px red}</style><div id="part-${part}">%TEMPLATE%</div>`
+            };
+          })
+        }
+      },
+      constants: [
+        { type: 'attribute', name: 'clearable', value: true },
+        { type: 'attribute', name: 'value', value: 'value' },
+        { type: 'attribute', name: 'label', value: 'Label' },
+        { type: 'attribute', name: 'help-text', value: 'help-text' },
+        {
+          type: 'slot',
+          name: 'left',
+          value: '<sd-icon slot="left" library="global-resources" name="system/picture"></sd-icon>'
+        },
+        {
+          type: 'slot',
+          name: 'right',
+          value: '<sd-icon slot="right" library="global-resources" name="system/picture"></sd-icon>'
+        }
+      ],
+      args
+    });
+  }
+};
+
+/**
+ * `sd-input` is fully accessibile via keyboard.
+ */
+
+export const Mouseless = {
+  render: (args: any) => {
+    return html`<div class="mouseless w-[231px]">${generateTemplate({ args })}</div>`;
+  },
+
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const el = canvasElement.querySelector('.mouseless sd-input');
+    await waitUntil(() => el?.shadowRoot?.querySelector('input'));
+    el?.shadowRoot?.querySelector('input')!.focus();
+  }
+};
+
+/**
+ * Sample implementation of a currency stepper.
+ */
+
+export const Samples = {
+  parameters: {
+    controls: {
+      include: []
+    }
+  },
+  render: () => {
+    return html`
+      <style>
+        .headline {
+          margin-bottom: 16px;
+          padding: 16px;
+          background: #e0e0e0;
+          text-align: left;
+          font-size: 14px;
+          font-weight: bold;
+          width: 100%;
+          box-sizing: border-box;
+        }
+      </style>
```

Comment:
Could you move this into a decorator, so that we don't see it in the story's code? :)